### ISSUE TITLE
Orca save model can not save the optimizer status with spark backend

### ISSUE THREAD

**Issue by ForJadeForest** [intel-analytics/BigDL#7268]
I try to resume training the model with the spark backend, but when loading the model the learning rate is the initial learning rate.
I find the the function `get_model()` of spark estimator only create a new model and load the weights. So I think it's better to load the optimizer status in the function.
```python

    def get_model(self,
                  set_weights: bool=True) -> "Model":
        """
        Returns the learned model.

        :return: the learned model.
        """
        if self.model_creator is not None:
            model = self.model_creator(self.config)
        else:
            model = load_model(**self.load_params)

        if set_weights and self.model_weights is not None:
            model.set_weights(self.model_weights)
        return model

```

**Comment by hkvision** [comment]
@sgwhat Please take a look

**Comment by sgwhat** [comment]
Does there exist any better method to load optimizer status?

**Comment by hkvision** [comment]
Can spark backend follow the implementation of ray backend?

**Comment by sgwhat** [comment]
> Can spark backend follow the implementation of ray backend?

No, ray backend could save the model directly from rank 0 worker, but pyspark backend cannot, which needs to re-constrct a model and load pre-training status.

**Comment by hkvision** [comment]
Does this post help? https://stackoverflow.com/questions/49503748/save-and-load-model-optimizer-state

**Comment by sgwhat** [comment]
> Does this post help? https://stackoverflow.com/questions/49503748/save-and-load-model-optimizer-state

Since we need to re-construct a new model, that's not a good way to `model.optimizer.get_state()`. But maybe I could add an extra process that would return optimizer states after `model.fit` call, then we could use `model.optimizer.load_state()` in `get_model()` call?

**Comment by hkvision** [comment]
Can we simulate how we do with weights (broadcast and put it in the estimator) for the optimizer states?

**Comment by sgwhat** [comment]
> Can we simulate how we do with weights (broadcast and put it in the estimator) for the optimizer states?

yes we could.

**Comment by sgwhat** [comment]
Currently, we can only support tensorflow 2.10.0 and previous version.

**Comment by hkvision** [comment]
https://github.com/intel-analytics/BigDL/pull/7555#issuecomment-1439931772
ray backend doesn't have this issue since re-creating model is not involved.
